core/dbt/task/retry.py

```diff
@@ -0,0 +1,140 @@
+from dbt.flags import get_flags
+from dbt.graph import GraphQueue
+from dbt.task.base import ConfiguredTask
+from dbt.task.build import BuildTask
+from dbt.task.compile import CompileTask
+from dbt.task.freshness import FreshnessTask
+from dbt.task.generate import GenerateTask
+from dbt.task.run import RunTask
+from dbt.task.run_operation import RunOperationTask
+from dbt.task.seed import SeedTask
+from dbt.task.show import ShowTask
+from dbt.task.snapshot import SnapshotTask
+from dbt.task.test import TestTask
+
+RETRYABLE_STATUSES = {NodeStatus.Error, NodeStatus.Fail, NodeStatus.Skipped, NodeStatus.RuntimeErr}
+
+TASK_DICT = {
+    "build": BuildTask,
+    "compile": CompileTask,
+    "freshness": FreshnessTask,
+    "generate": GenerateTask,
+    "run": RunTask,
+    "seed": SeedTask,
+    "show": ShowTask,
+    "snapshot": SnapshotTask,
+    "test": TestTask,
+    "run-operation": RunOperationTask,
+}
+
+CMD_DICT = {
+    "build": CliCommand.BUILD,
+    "compile": CliCommand.COMPILE,
+    "freshness": CliCommand.SOURCE_FRESHNESS,
+    "generate": CliCommand.DOCS_GENERATE,
+    "run": CliCommand.RUN,
+    "seed": CliCommand.SEED,
+    "show": CliCommand.SHOW,
+    "snapshot": CliCommand.SNAPSHOT,
+    "test": CliCommand.TEST,
+    "run-operation": CliCommand.RUN_OPERATION,
+}
+
+
+class RetryTask(ConfiguredTask):
+    def __init__(self, args, config, manifest):
+        super().__init__(args, config, manifest)
+
+        state_path = self.args.state or self.config.target_path
+
+        if self.args.warn_error:
+            RETRYABLE_STATUSES.add(NodeStatus.Warn)
+
+        self.previous_state = PreviousState(
+            state_path=Path(state_path),
+            target_path=Path(self.config.target_path),
+            project_root=Path(self.config.project_root),
+        )
+
+        if not self.previous_state.results:
+            raise DbtRuntimeError(
+                f"Could not find previous run in '{state_path}' target directory"
+            )
+
+        self.previous_args = self.previous_state.results.args
+        self.previous_command_name = self.previous_args.get("which")
+        self.task_class = TASK_DICT.get(self.previous_command_name)
+
+    def run(self):
+        unique_ids = set(
+            [
+                result.unique_id
+                for result in self.previous_state.results.results
+                if result.status in RETRYABLE_STATUSES
+            ]
+        )
+
+        cli_command = CMD_DICT.get(self.previous_command_name)
+
+        bad_args = {
+            "show": lambda x: True,
+            "resource_types": lambda x: x == [],
+            "warn_error_options": lambda x: x == {"exclude": [], "include": []},
+        }
+
+        for k, v in bad_args.items():
+            if k in self.previous_args and v(self.previous_args[k]):
+                del self.previous_args[k]
+
+        retry_flags = Flags.from_dict(cli_command, self.previous_args)
+
+        retry_profile = load_profile(
```

Comment:
Why we need to reload the profile and project here? Given that the current runtime config already contains these information.
Is it just because we need to reconstruct the runtime_config?
If that's the case we should just overwrite attributes on the current runtime config with the values in retry flags.